aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -184,7 +187,7 @@ public Completable load() {
         return Completable.create(emitter -> {
             inFlightMutations.clear();
             mutationQueue.clear();
-            storage.query(PendingMutation.PersistentRecord.class, Where.matchesAll(),
+            storage.query(PendingMutation.PersistentRecord.class, Where.matchesAll().paginated(Page.firstPage()),
```

Comment:
Loading one page at a time will definitely help the memory pressure compared to loading the entire pending mutation table.  However an alternative which is used in the [Javascript implementation here](https://github.com/aws-amplify/amplify-js/blob/a8f4cbe3a238098a9d916b139fc5771c57149615/packages/datastore/src/sync/processors/mutation.ts#L164) is to pull the items one at a time for processing.  This would eliminate the need for `mutationQueue` altogether and would further reduce memory usage.

Reply:
The mutation processor intertwines with the subscription processor in ways that prevented me from straight deleting the mutationQueue in the time I had to make a solution.

I agree nuking it from orbit seems more profitable. Just haven't the margin at the moment and the changes as is have run offline and online for months straight on the testbed.

Reply:
Having some internal discussions about whether we want to accept this as a step in the right direction or wait till we can remove mutationqueue altogether.